File: day1/day1.jl
function split_input(file_name::String)::Vector{Vector{String}}

    buckets = []

    open(file_name) do f
        lines = readlines(f)
        bucket = []

        for line in lines
            if line == ""
                append!(buckets, bucket)
                bucket = []
            else
                #value = parse(Int, line)
                append!(bucket, value)
            end
        end
    end
    return buckets     
end

buckets = split_input("test_input.txt")

File: day1/day1.py
def split_file(file_name:str = "input.txt") -> list[int]:

    output = []

    with open(file_name, "r") as f:
        bucket = []

        for line in f:

            if line == "\n":
                if bucket != []:
                    output.append(bucket)
                    bucket = []
            else:
                bucket.append(int(line.strip()))
        if bucket != []:
            output.append(bucket)
            
    return output


def sum_buckets(buckets:list[list[int]]) -> list[int]:
    return list(map(sum, buckets))


if __name__ == "__main__":

    buckets = split_file()
    sums = sum_buckets(buckets)
    sums.sort(reverse=True)
    print("max = ", sums[0])
    print("top three = ", sum(sums[:3]))

